src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -150,6 +201,22 @@ class AWSNodeGroupInputVars(schema.Base):
     permissions_boundary: Optional[str] = None
     ami_type: Optional[AWSAmiTypes] = None
     launch_template: Optional[AWSNodeLaunchTemplate] = None
+    node_taints: list[dict]
+
+    @field_validator("node_taints", mode="before")
```

Comment:
This code is repeated for GCP and AWS NodeGroupInputVars classes, but that's b/c the format expected by GCP and AWS terraform modules for taints happens to be the same.  I think they could evolve separately and so I chose to duplicate the code in this case.